include/powerloader/curl.hpp

```diff
@@ -35,17 +36,25 @@ namespace powerloader
         bool m_serious;
     };
 
-    struct POWERLOADER_API Response
+    struct POWERLOADER_API BaseResponse
     {
-        std::map<std::string, std::string> header;
-        mutable std::stringstream content;
+        std::map<std::string, std::string> headers;
 
-        curl_off_t avg_speed;
+        curl_off_t average_speed;
         curl_off_t downloaded_size;
         long http_status;
         std::string effective_url;
 
         bool ok() const;
+
+        tl::expected<std::string, std::out_of_range> get_header(const std::string& header) const;
+
+        void fill_values(CURLHandle& handle);
+    };
+
+    struct POWERLOADER_API Response : public BaseResponse
```

Comment:
I'm not sure I understand why the split in hierarchy is necessary here?

If the goal is to not always have/provide all the data, I would avoid inheritance and just have another structure with the additional data which would be stored in an optional or a unique/shared_ptr, or `std::any` (depending on how copying `Response` is supposed to work, how it's passed to the public interface etc.).

Reply:
DOne!